ENT-6790: Updated the design of the executive eduction course landing page.

saleem-latif opened the pull request:
__Jira Ticket:__ [ENT-6790](https://2u-internal.atlassian.net/browse/ENT-6790)

__Description:__
As part of PLI (Product Line Integration) enrollment flow with existing ecommerce system, an enterprise learner with an enterprise offer should see the course registration page with applicable discounts and course registration summary.

**Acceptance Criteria:**

1. Registration page with mocked applicable discounts and course registration summary is available in the enterprise learner portal
2. Integrate with backend and test the flow
3. DOB is collected on this page.
4. User accepts both data sharing consent and terms and conditions. Both are required to complete the registration.
5. If the enterprise’s data sharing consent flag is False, hide data sharing consent (it is not required to complete registration).
6. If the user’s DOB places them less than 18 years old, they cannot complete the enrollment (see second mock above).

Here is the screenshot of the page after the updates.
![screencapture-localhost-8734-test-enterprise-executive-education-2u-2023-04-06-18_12_57](https://user-images.githubusercontent.com/7114956/230390584-9ba8a1e7-1aed-4f59-a681-8d77e3d8d95a.png)

# For all changes

- [x] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [x] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

zamanafzal reviewed (approved):
LGTM

jajjibhai008 reviewed (approved):
LGTM 👍

muhammad-ammar commented on src/components/executive-education-2u/components/CourseSummaryCard.jsx:
```diff
@@ -0,0 +1,71 @@
+                <Col className="small font-weight-light text-gray-500 justify-content-start">Course total:</Col>
+                <Col className="justify-content-end">
+                  <Row className="justify-content-end margin-right-10">
+                    ${courseMetadata.priceDetails?.price ? `${numberWithPrecision(courseMetadata.priceDetails.price)} ${courseMetadata.priceDetails.currency}` : 'Unknown'}
```
Is there are possibility that `courseMetadata.priceDetails?.price` will be `falsy`?

saleem-latif replied:
I was getting an error for some cases, that is why I added this check.

bilaltahir21 reviewed (approved):
LGTM :shipit:

saleem-latif commented:
@adamstankiewicz I have incorporated your feedback changes, please take another look.

saleem-latif commented:
I have also added back `firstName` and the `lastName` fields back into the form.

adamstankiewicz commented on src/components/executive-education-2u/UserEnrollmentForm.jsx:
```diff
@@ -70,6 +80,7 @@ const UserEnrollmentForm = ({
           dateOfBirth: values.dateOfBirth,
         },
         termsAcceptedAt: toISOStringWithoutMilliseconds(new Date(Date.now()).toISOString()),
+        ...(enableDataSharingConsent ? { dataShareConsent: true } : {}),
```
[curious] Should we be relying on `values.dataSharingConsent` here rather than always hardcoding to `true`?

adamstankiewicz commented on src/components/executive-education-2u/components/CourseSummaryCard.jsx:
```diff
@@ -0,0 +1,76 @@
+} from '@edx/paragon';
+import { numberWithPrecision } from '../../course/data/utils';
+
+const CourseSummaryCard = ({ courseMetadata, enrolmentCompleted }) => (
```
nit: typo in `enrolmentCompleted` -> `enrollmentCompleted`

saleem-latif commented:
@adamstankiewicz I have incorporated your feedback. I have added a comment regarding the age validation logic. Let me know if you agree, otherwise I will move age validation into the form validation logic.

adamstankiewicz commented on src/components/executive-education-2u/ExecutiveEducation2UPage.jsx:
```diff
@@ -83,25 +102,45 @@ const ExecutiveEducation2UPage = () => {
+            {isLoading || !contentMetadata ? (
               <Skeleton containerTestId="loading-skeleton-page-title" />
-            ) : pageTitle}
+            ) : 'Your Registration(s)'}
```
nit: casing is off; should be "Your registration(s)": 

<img width="247" alt="image" src="https://user-images.githubusercontent.com/2828721/231522258-e3b6d7a6-c8b0-4db5-908c-3f3d7eecdd8c.png">

adamstankiewicz commented on src/components/executive-education-2u/ExecutiveEducation2UPage.jsx:
```diff
@@ -45,30 +50,44 @@ const ExecutiveEducation2UPage = () => {
+  }
+
   return (
     <Container size="lg" className="py-5">
```
The mocks depict a background color (i.e., `.bg-light-200`) for the PLI enrollment flow. Was this intentionally left out or should it be included? If it should be included, it could be implemented by wrapping this `Container` with:

```jsx
<div className="bg-light-200">
  <Container>{...}</Container>
</div>
```

<img width="1480" alt="image" src="https://user-images.githubusercontent.com/2828721/231522887-d8b7764a-0eab-4c37-91ed-98efe50139be.png">

adamstankiewicz commented on src/components/executive-education-2u/components/RegistrationSummaryCard.jsx:
```diff
@@ -0,0 +1,56 @@
+                    ${priceDetails?.price ? `${String(0).padStart(priceDetails.price.toString().length, '0') }.00` : '0.00'} {priceDetails?.currency ? priceDetails.currency : CURRENCY_USD}
+                  </Row>
+                  <Row className="justify-content-end small font-weight-light text-gray-500 mr-2.5">
+                    Tax Included
```
nit: casing. Mocks have this as "Tax **i**ncluded"

<img width="542" alt="image" src="https://user-images.githubusercontent.com/2828721/231522976-f34392d0-2640-43b5-b32f-8a4d343f6ce5.png">

adamstankiewicz commented on src/components/executive-education-2u/styles/_ExecutiveEducation2UPage.scss:
```diff
@@ -0,0 +1,34 @@
+.course-summary {
+  .registration-outline {
```
These styles don't appear to be applying. I would also recommend instead using `.bg-light-500`, `.p-3`, and `.rounded-lg` on the `div` where this class name is applied instead of defining custom styles.

adamstankiewicz commented on src/components/executive-education-2u/styles/_ExecutiveEducation2UPage.scss:
```diff
@@ -0,0 +1,34 @@
+.registration-summary {
+  .registration-details {
+    border: 1px solid $gray-200;
+    border-radius: 10px;
```
Use `.rounded-lg` instead.

adamstankiewicz commented on src/components/executive-education-2u/styles/_ExecutiveEducation2UPage.scss:
```diff
@@ -0,0 +1,34 @@
+}
+.registration-summary {
+  .registration-details {
+    border: 1px solid $gray-200;
```
Use `.border` instead.

adamstankiewicz commented on src/components/executive-education-2u/styles/_ExecutiveEducation2UPage.scss:
```diff
@@ -0,0 +1,34 @@
+  }
+
+  .age-error-message {
+    border: 1px solid $light-300;
```
Use `.border` and `.p-3` instead of defining custom styles to do the same.

adamstankiewicz commented on src/components/executive-education-2u/components/CourseSummaryCard.jsx:
```diff
@@ -0,0 +1,76 @@
+          </Col>
+          <Col xs={12} lg={{ span: 4, offset: 0 }}>
+            <div className="course-details">
+              <Row>
+                <Col className="small font-weight-light text-gray-500 justify-content-start">{enrollmentCompleted ? 'Start date:' : 'Available start date:'}</Col>
+                <Col className="justify-content-end"><Row className="justify-content-end mr-2.5">{courseMetadata.startDate}</Row></Col>
+              </Row>
+              <Row>
```
It looks like the alignment of the values is off from their labels:

<img width="1181" alt="image" src="https://user-images.githubusercontent.com/2828721/231525732-aa6bde5e-d11b-4d38-bcf6-ebc0b5434d7f.png">

From Figma mocks:

<img width="627" alt="image" src="https://user-images.githubusercontent.com/2828721/231525792-d3c32374-1b13-45e5-be6e-cf760081e213.png">

saleem-latif replied:
Strange, Alignment is looking good on my end, I tested on Chrome, Firefox and Safari.
<img width="1213" alt="Screen Shot 2023-04-13 at 11 41 10 AM" src="https://user-images.githubusercontent.com/7114956/231675544-deaec526-8e71-46bf-a54c-5bde50c272c8.png">

saleem-latif replied:
I can not find any issue in the code as well as to why it would happen on some machines. Can you try again? or maybe suggest a solution if you can fix it on your end?

adamstankiewicz replied:
Interesting. Just pulled down the branch again to try again. I still see the alignment issue, at least when using the `@edx/brand-edx.org` theme, in Chrome. That said, adding an `align-items-center` class name on these `Row` components seems to fix it.

```jsx
<Row className="align-items-center">
```

<img width="1183" alt="image" src="https://user-images.githubusercontent.com/2828721/232473865-7a257e3d-6417-4fe7-9bf3-821540fe8e90.png">

adamstankiewicz commented on src/components/executive-education-2u/components/CourseSummaryCard.jsx:
```diff
@@ -0,0 +1,76 @@
+              </Row>
+              <Row>
+                <Col className="small font-weight-light text-gray-500 justify-content-start">Course duration:</Col>
+                <Col className="justify-content-end"><Row className="justify-content-end mr-2.5">{courseMetadata.duration}</Row></Col>
```
nit: casing. Mocks show "weeks" versus implementation shows "Week". Should be plural when appropriate.

adamstankiewicz commented on src/components/executive-education-2u/components/RegistrationSummaryCard.jsx:
```diff
@@ -0,0 +1,56 @@
+              <Row>
+                <Col xs={12} lg={{ span: 6, offset: 0 }} className="small font-weight-light text-gray-500 justify-content-start">Registration total:</Col>
+                <Col xs={12} lg={{ span: 6, offset: 0 }} className="justify-content-end">
+                  <Row className="justify-content-end mr-2.5">
```
[clarification] Typically it doesn't really make sense to render a `Row` without any `Col` children. Could this `Row` (and the similar ones below) be replaced with a `div` instead?

Example:
```jsx
<div className="d-flex justify-content-end mr-2.5">{...}</div>
```

adamstankiewicz commented on src/components/executive-education-2u/UserEnrollmentForm.jsx:
```diff
@@ -123,138 +135,194 @@ const UserEnrollmentForm = ({
+                <Row className="justify-content-end">
+                  {
+                    isAgeValid && (
+                      <StatefulButton
```
The `StatefulButton` is not properly aligned with the rest of the page/form:

<img width="1246" alt="image" src="https://user-images.githubusercontent.com/2828721/231528156-4f18b27a-8286-4dc1-9fea-0e56be215000.png">

Note how it extends beyond the width of the "Registration summary" card. I would have expected the button to be flush with the rightmost edge of the "Registration summary" card for consistent alignment.

adamstankiewicz commented on src/components/executive-education-2u/EnrollmentCompleted.jsx:
```diff
@@ -0,0 +1,45 @@
+    return <Redirect to={`/${enterpriseConfig.slug}`} />;
+  }
+  return (
+    <Container size="md" className="py-5">
```
[curious/clarification] Is this supposed to have the same `Container` `size` as the registration form page (i.e., `size="lg"`?

adamstankiewicz commented on src/components/executive-education-2u/UserEnrollmentForm.jsx:
```diff
@@ -123,138 +135,194 @@ const UserEnrollmentForm = ({
+                    <Row className="mb-4 age-error-message">
+                      <Col xs={12} lg={12}>
+                        <p className="font-italic">
+                          Unfortunately you don&apos;t meet minimum age requirement and without any parent or guardian
```
typo: "...meet _the_ minimum age requirement"

adamstankiewicz commented on src/components/executive-education-2u/UserEnrollmentForm.jsx:
```diff
@@ -123,138 +135,194 @@ const UserEnrollmentForm = ({
+                      <Col xs={12} lg={12}>
+                        <p className="font-italic">
+                          Unfortunately you don&apos;t meet minimum age requirement and without any parent or guardian
+                          consent you can not proceed with your registration.
```
typo: according to mocks, "can not" -> "cannot"

saleem-latif commented:
@adamstankiewicz I have incorporated your feedback changes, please take another look.
